### PR TITLE
Clarify that ::foo paths are not necessarily based off of the "crate root"

### PR DESCRIPTION
**Pull request by Manishearth**
Part of https://github.com/rust-lang/rust/issues/82876

### CODE REVIEW

File: src/paths.md [hunk 1]
```diff
@@ -159,16 +159,17 @@ how it is resolved.
...
+the path must resolve to an item.
+
+> **Edition Differences**: In the 2015 Edition, identifiers resolve from the "crate root"
+> (`crate::` in the 2018 edition), which contains a variety of different items, including
+> external crates, default crates such as `std` or `core`, and items in the top level of
+> the crate (including `use` imports).
 >
-> Beginning with the 2018 Edition, paths starting with `::` can only reference
-> crates in the [extern prelude].
+> Beginning with the 2018 Edition, paths starting with `::` resolve from from
```
Comment (CryZe) [inline src/paths.md:171]:
```suggestion
> Beginning with the 2018 Edition, paths starting with `::` resolve from
```

File: src/paths.md [hunk 2]
```diff
@@ -159,16 +159,17 @@ how it is resolved.
...
 ### `::`
 
-Paths starting with `::` are considered to be global paths where the segments of the path
-start being resolved from the crate root. Each identifier in the path must resolve to an
-item.
-
-> **Edition Differences**: In the 2015 Edition, the crate root contains a variety of
-> different items, including external crates, default crates such as `std` or `core`, and
-> items in the top level of the crate (including `use` imports).
+Paths starting with `::` are considered to be "global paths" where the segments of the path
```
Comment (Havvy) [inline src/paths.md:162]:
```suggestion
Paths starting with `::` are considered to be *global paths* where the segments of the path
```

File: src/paths.md [hunk 3]
```diff
@@ -159,16 +159,17 @@ how it is resolved.
...
+
+> **Edition Differences**: In the 2015 Edition, identifiers resolve from the "crate root"
+> (`crate::` in the 2018 edition), which contains a variety of different items, including
+> external crates, default crates such as `std` or `core`, and items in the top level of
+> the crate (including `use` imports).
 >
-> Beginning with the 2018 Edition, paths starting with `::` can only reference
-> crates in the [extern prelude].
+> Beginning with the 2018 Edition, paths starting with `::` resolve from
+> crates in the [extern prelude], i.e. they _must_ be followed by the name of a crate.
```
Comment (Havvy) [inline src/paths.md:172]:
```suggestion
> crates in the [extern prelude]. That is, they must be followed by the name of a crate.
```

Trying to keep italics for defining terms, not for emphasizing specific words. If you think we absolutely **must** emphasize the word, we could make it bold, but I don't think we do.

**Review by Havvy — CHANGES_REQUESTED** [review]

**Review by Havvy — CHANGES_REQUESTED** [review]
Also, I think the default is to describe the currentmost edition's behavior outside of the edition differences section and specify how it's different in older in the edition differences section. So the part that is in the "Beginning with the 2018 edition" should be the description of what a global path is.

**Comment by ehuss** [comment]
>  I think the default is to describe the currentmost edition's behavior outside of the edition differences section

I don't think this issue has been settled (see #698). The original intent was to have the main text describe what is common in both editions, and to use sidebars to explain where it diverges (with exceptions like the async chapters).  I'm starting to lean a little more towards preferring the "describe latest edition" approach, but there are some places where it can be a little awkward. 

Maybe @Manishearth has an opinion on which approach seems better overall?

(BTW, thanks for the PR, I kinda made a mess on this section.)

**Comment by Manishearth** [comment]
When I wrote this PR I thought about it and decided not to because it was the reference, but I don't mind doing it. I did find it _easier_ to describe what it _was_ (a "global path") and then describe edition differences in a clear mirrored fashion.

**Review by Havvy — APPROVED** [review]

**Comment by Havvy** [comment]
💟 Thanks!

I still prefer the idea of documenting the current edition as primary, but that can be redecided on elsewhere. The whole await/async thing might actually be in a better spot if `k#keywords` become a thing.

**Pull request merged**